File: OJ/usaco/s1/s12/transform.pas
{
ID:lijianx1
PROG:transform
LANG:PASCAL
}
program transform;
const maxn=10;
var a,tmp,s:array[1..maxn,1..maxn] of boolean;
    i,j,n:longint;
    c:char;

procedure change(k:longint);
 var i,j:longint;
 begin
     for i:=1 to n do
        for j:=1 to n do
           case k of
               1:tmp[j,n+1-i]:=a[i,j];
               2:tmp[n+1-i,n+1-j]:=a[i,j];
               3:tmp[n+1-j,i]:=a[i,j];
               4:tmp[i,n+1-j]:=a[i,j];
               5:tmp[n+1-j,n+1-i]:=a[i,j];
               6:tmp[n+1-i,j]:=a[i,j];
               7:tmp[j,i]:=a[i,j];
               8:tmp[i,j]:=a[i,j];
           end;
 end;

procedure check(k:longint);
 var i,j:longint;
 begin
     for i:=1 to n do
       for j:=1 to n do
         if tmp[i,j]<>s[i,j] then exit;
     writeln(k);
     close(output);
     halt;
 end;

begin
    assign(input,'transform.in'); reset(input);
    readln(n);
    for i:=1 to n do begin
        for j:=1 to n do begin
	    read(c);
	    a[i,j]:=c='@';
        end;
        readln;
    end;
    for i:=1 to n do begin
        for j:=1 to n do begin
            read(c);
            s[i,j]:=c='@';
        end;
        readln;
    end;
    close(input);
    assign(output,'transform.out'); rewrite(output);
    change(1); check(1);
    change(2); check(2);
    change(3); check(3);
    change(4); check(4);
    change(5); check(5);
    change(6); check(5);
    change(7); check(5);
    change(8); check(6);
    writeln(7);
    close(output);
end.

File: OJ/usaco/cpp/transform.cpp
/*
ID:oifox20071
LANG:C++
PROG:transform
*/
#include <iostream>
using namespace std;
const
	int maxn = 10;
bool start[maxn][maxn], target[maxn][maxn], temp[maxn][maxn];
int n;

int change(int x)
{
	for (int i = 0; i < n; ++ i)
		for (int j = 0; j < n; ++ j)
			switch (x)
			{
				case 1: temp[j][n - i - 1] = start[i][j]; break;
				case 2: temp[n - i - 1][n - j - 1] = start[i][j]; break;
				case 3: temp[n - j - 1][i] = start[i][j]; break;
				case 4: temp[i][n - j - 1] = start[i][j]; break;
				case 5: temp[n - j - 1][n - i - 1] = start[i][j]; break;
				case 6: temp[n - i - 1][j] = start[i][j]; break;
				case 7: temp[j][i] = start[i][j];
				case 8: temp[i][j] = start[j][i];
			}
	return 0;
}

int check(int x)
{
	for (int i = 0; i < n; ++ i)
		for (int j = 0; j < n; ++ j)
			if (temp[i][j] != target[i][j]) return 0;
	cout << x << endl;
	exit(0);
}

int main()
{
	freopen("transform.in", "r", stdin); freopen("transform.out", "w", stdout);
	cin >> n;
	char c;
	for (int i = 0; i < n; ++ i)
		for (int j = 0; j < n; ++ j)
		{
			cin >> c;
			start[i][j] = c == '@';
		}
	for (int i = 0; i < n; ++ i)
		for (int j = 0; j < n; ++ j)
		{
			cin >> c;
			target[i][j] = c == '@';
		}
	change(1); check(1);
	change(2); check(2);
	change(3); check(3);
	change(4); check(4);
	change(5); check(5);
	change(6); check(5);
	change(7); check(5);
	change(8); check(6);
	cout << 7 << endl;
	return 0;
}
